lib/active_record/connection_adapters/abstract_mysql.rb

```diff
@@ -1,21 +1,19 @@
-# This module fails in Rails 7.0 becuase the method being modified has been
-# changed to a class method.
-
 module ActiveRecord
   module ConnectionAdapters
     class AbstractMysqlAdapter
-      def initialize_type_map_with_enum(m = type_map)
-        initialize_without_enum(m)
-        register_enum_type(m)
+      class << self
+        def register_enum_type(mapping)
+          mapping.register_type(%r(enum)i) do |sql_type|
+            Type::Enum.new(limit: sql_type.scan(/'(.*?)'/).flatten)
```

Comment:
Do you understand this limit code? I know it's copy/pasted, but the `limit` is quite opaque to me.

Reply:
The third party tool does this: `limit = sql_type.sub(/^enum\('(.+)'\)/i, '\1').split("','").map { |v| v.to_sym }` which is actually different (our code ends up with strings, the other goes into symbols)

I think the `sql_type` comes in here as a string formatted like `enum('low', 'normal', 'high')` and this is pulling the enum values into a Ruby array. I'm not 100% sure why you need to do that, but I think it's what makes the ActiveRecord conversion work.